app/screens/route-details/route-details-screen.tsx

```diff
@@ -49,6 +57,17 @@ export const RouteDetailsScreen = observer(function RouteDetailsScreen({ route }
   const openFirstRideAlertSheet = () => {
     bottomSheetRef.current?.expand()
   }
+  const eventTitle = translate("plan.trainFromToStation", {
+    trainNumber: route.params.routeItem.trains[0].trainNumber,
+    origin: route.params.routeItem.trains[0].originStationName,
+    destination: route.params.routeItem.trains[0].destinationStationName,
+  })
+  const eventConfig: CreateOptions = {
+    title: eventTitle,
+    startDate: new Date(route.params.routeItem.departureTime).toISOString(),
+    endDate: new Date(route.params.routeItem.arrivalTime).toISOString(),
+    notes: eventTitle,
+  }
```

Comment:
Extract this to a function outside the component, and pass the `routeItem` as an argument.

Reply:
seem to already have such argument (as Memo...).